### PR TITLE
added job tests

### PR DESCRIPTION
**Pull request by MichaelClifford**
Here is the Job tests added so far.

### CODE REVIEW

File: tests/unit_test.py [hunk 1]
```diff
@@ -1575,3 +1586,126 @@ def test_cmd_line_generation():
...
+    assert type(job_info._scheduler) == type(str())
+    return ddp_job
+
+
+def test_DDPJob_status(mocker):
+    ddp_job = test_DDPJob_creation(mocker)
+    mocker.patch(
+        "codeflare_sdk.job.jobs.torchx_runner.status", return_value="fake-status"
+    )
+    assert len(torchx_runner.status.mock_calls) == 0
```
Comment (KPostOffice) [inline tests/unit_test.py]: What is this line doing? Also, you can replace this with
```
torchx_runner.status.assert_not_called()
```

File: tests/unit_test.py [hunk 2]
```diff
@@ -1575,3 +1586,126 @@ def test_cmd_line_generation():
...
+    assert ddp.j == "2x1"
+    assert ddp.env == {"test": "test"}
+    assert ddp.max_retries == 0
+    assert ddp.mounts == []
+    assert ddp.rdzv_port == 29500
+    assert ddp.scheduler_args == {"requirements": "test"}
+    return ddp
+
+
+def test_DDPJobDefinition_dry_run():
```
Comment (KPostOffice) [inline tests/unit_test.py:1639]: When you change this to check some values, can you also add a test which makes sure that cluster values are used if no value is provided.

**Review by KPostOffice — COMMENTED** [review]
Looks good. Just a couple comments

File: tests/unit_test.py (hunk 1 above)
Comment (MichaelClifford) [reply tests/unit_test.py]: lol, yeah, this does nothing. Changing it to something like this so we confirm that the value for `self. _app_handle` "fake-dahsboard-url" is passed to the functions. 

```
 mocker.patch("codeflare_sdk.job.jobs.torchx_runner.log_lines", 
         return_value="fake-logs")
    assert ddp_job.logs() == "fake-logs"
    _, args, kwargs = torchx_runner.log_lines.mock_calls[0]
    assert args[0] == "fake-dashboard-url"
    
```

Comment (KPostOffice) [reply tests/unit_test.py]: same with below

**Comment by MichaelClifford** [comment]
Changes made @KPostOffice, Let me know what you think. Thanks :)

**Review by KPostOffice — APPROVED** [review]
/lgtm

**Pull request merged**